nexus/external-api/src/lib.rs

```diff
@@ -1568,7 +1910,7 @@ pub trait NexusExternalApi {
     async fn networking_bgp_announce_set_update(
         rqctx: RequestContext<Self::Context>,
         config: TypedBody<params::BgpAnnounceSetCreate>,
-    ) -> Result<HttpResponseCreated<BgpAnnounceSet>, HttpError>;
+    ) -> Result<HttpResponseOk<BgpAnnounceSet>, HttpError>;
```

Comment:
Our current test framework expects `put` to return `HttpResponseOk`. I think this could / should be changed, but that is out of scope for this PR.

https://github.com/oxidecomputer/omicron/blob/122eccf9a76749776b91a42f81735b6f0ec40f5a/nexus/test-utils/src/http_testing.rs#L652-L666